mill-db-repo/src/test/java/org/duracloud/mill/manifest/jpa/ManifestStoreIntegrationTest.java

```diff
@@ -0,0 +1,175 @@
+        assertEquals(account, item.getAccount());
+        assertEquals(storeId, item.getStoreId());
+        assertEquals(spaceId, item.getSpaceId());
+        assertEquals(contentId, item.getContentId());
+        assertEquals(contentChecksum, item.getContentChecksum());
+        assertEquals(contentMimetype, item.getContentMimetype());
+        assertEquals(contentSize, item.getContentSize());
+        assertEquals(timestamp, item.getModified());
+    }
+
+    private ManifestItem addAndRetrieveItem(Date timestamp) throws ManifestItemWriteException, NotFoundException {
+        ManifestStore store = getStore();
+        assertTrue(store.addUpdate(account, storeId, spaceId, contentId, contentChecksum, contentMimetype,
+                contentSize,
+                timestamp));
+        ManifestItem item = store.getItem(account, storeId, spaceId, contentId);
+        return item;
+    }
+
+    @Test
+    public void testUpdate() throws Exception {
+
+        long oneSecondAgo = System.currentTimeMillis() - 1000;
+        Date timestamp = new Date(oneSecondAgo);
+        ManifestItem item1 = addAndRetrieveItem(timestamp);
+        assertEquals(0, item1.getVersion());
+
+        ManifestStore store = getStore();
+
+        // change the checksum
+        String contentChecksum2 = "new checksum";
+        String contentMimetype2 = "contentMimetype2";
+        String contentSize2 = "contentSize2";
+        long ms999ago = oneSecondAgo + 1;
+
+        assertTrue(store.addUpdate(account, storeId, spaceId, contentId, contentChecksum2, contentMimetype2,
+                contentSize2,
+                new Date(ms999ago)));
+
+        // get it
+        ManifestItem item2 = store.getItem(account, storeId, spaceId, contentId);
+
+        // verify time modified exactly matches new timestamp
+        assertEquals(ms999ago, item2.getModified().getTime());
+        // verify updated fields
+        assertEquals(contentChecksum2, item2.getContentChecksum());
+        assertEquals(contentMimetype2, item2.getContentMimetype());
+        assertEquals(contentSize2, item2.getContentSize());
+
+        // verify version autoincrement
+        assertEquals(item2.getVersion(), item1.getVersion() + 1);
+    }
+
+    @Test
+    public void testIgnoreUpdateDueToOutOfOrderMessage() throws Exception {
+
+        long currentTime = System.currentTimeMillis();
+        Date timestamp = new Date(currentTime);
+        ManifestItem item1 = addAndRetrieveItem(timestamp);
+        ManifestStore store = getStore();
+
+        // change the checksum
+        long pastTime = currentTime - 1;
+        String newChecksum = "checksum-2";
+        assertFalse(store.addUpdate(account, storeId, spaceId, contentId, newChecksum, contentMimetype,
+                contentSize,
+                new Date(pastTime)));
+
+        // get it
+        ManifestItem item2 = store.getItem(account, storeId, spaceId, contentId);
+
+        // verify time modified exactly matches new timestamp
```

Comment:
Looks like this is verifying that the time matches the original timestamp (rather than the new one). Code is right, comment could use an update.